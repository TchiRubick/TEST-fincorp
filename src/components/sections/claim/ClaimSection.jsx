import React from 'react';

import './ClaimSection.scss';

import Container from '../../container/Container';

import ablue from '../../../assets/img/after-blue.png';
import apink from '../../../assets/img/after-pink.png';

const ClaimSection = () => (
  <Container>
    <div className="section-claim">
      <div className="section-claim__title">
        <span className="section-claim__title-main">
          Simplified Claims
        </span>
        <span className="section-claim__title-sub">
          Easy as One, Two, Three
        </span>
      </div>
      <div className="section-claim__content">
        <div className="section-claim__content-item">
          <div className="section-claim__content-item__avatar __blue-circle">1</div>
          <div className="section-claim__content-item__title">
            Report Claim
          </div>
          <div className="section-claim__content-item__text">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            Proin quis nunc vitae velit rutrum suscipit non et eros.
          </div>
        </div>
        <div className="section-claim__content__separator">
          <img src={ablue} alt="after-blue" />
        </div>
        <div className="section-claim__content-item">
          <div className="section-claim__content-item__avatar __pink-circle">2</div>
          <div className="section-claim__content-item__title">
            Report Claim
          </div>
          <div className="section-claim__content-item__text">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            Proin quis nunc vitae velit rutrum suscipit non et eros.
          </div>
        </div>
        <div className="section-claim__content__separator">
          <img src={apink} alt="after-pink" />
        </div>
        <div className="section-claim__content-item">
          <div className="section-claim__content-item__avatar __green-circle">3</div>
          <div className="section-claim__content-item__title">
            Final Settlement
          </div>
          <div className="section-claim__content-item__text">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            Proin quis nunc vitae velit rutrum suscipit non et eros.
          </div>
        </div>
      </div>
    </div>
  </Container>
);

export default ClaimSection;
