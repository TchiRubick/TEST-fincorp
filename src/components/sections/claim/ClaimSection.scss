@import '../../style.scss';

.section-claim {

  .section-claim__title {
    text-align: center;
    display: flex;
    flex-direction: column;
    margin: 2rem auto;

    .section-claim__title-main {
      @include title;
      margin: 0 auto 1rem;
    }
    .section-claim__title-sub {
      color: gray;
      font-size: 13px;
    }
  }

  .section-claim__content {
    text-align: center;
    margin: auto;
    display: flex;
    justify-content: space-between;
    
    .section-claim__content__separator {
      align-self: center;
    }

    .section-claim__content-item {
      max-width: 10rem;
      text-align: center;


      .section-claim__content-item__avatar {
        border-radius: 50%;
        object-fit: scale-down;
        box-sizing: border-box;
        width: 5rem;
        height: 5rem;
        margin: 1rem auto;
        font-size: 3rem;
        padding-top: 0.8rem;

        &.__blue-circle {
          background-color: #b0f3ff;
          color: #99e1ee;
        }

        &.__pink-circle {
          background-color: #fbd1fe;
          color: #e9b4ed;
        }

        &.__green-circle {
          background-color: #daffd3;
          color: #bae7b0;
        }
      }

      .section-claim__content-item__title {
        font-size: 13px;
        font-weight: bold;
      }

      .section-claim__content-item__text {
        font-size: 11px;
        color: $color-subtext;
      }
    }
  }

}