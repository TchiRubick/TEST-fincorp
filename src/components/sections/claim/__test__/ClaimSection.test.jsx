import React from 'react';
import ReactDom from 'react-dom';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import ClaimSection from '../ClaimSection';

afterEach(cleanup);

it('renders good', () => {
  const div = document.createElement('div');

  ReactDom.render(<ClaimSection />, div);
});

it('as expected snapshot 1', () => {
  const tree = renderer.create(<ClaimSection />).toJSON();
  expect(tree).toMatchSnapshot();
});
