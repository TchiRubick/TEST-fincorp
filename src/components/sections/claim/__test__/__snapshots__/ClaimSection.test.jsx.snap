// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as expected snapshot 1 1`] = `
<div
  className="comp-container"
>
  <div
    className="section-claim"
  >
    <div
      className="section-claim__title"
    >
      <span
        className="section-claim__title-main"
      >
        Simplified Claims
      </span>
      <span
        className="section-claim__title-sub"
      >
        Easy as One, Two, Three
      </span>
    </div>
    <div
      className="section-claim__content"
    >
      <div
        className="section-claim__content-item"
      >
        <div
          className="section-claim__content-item__avatar __blue-circle"
        >
          1
        </div>
        <div
          className="section-claim__content-item__title"
        >
          Report Claim
        </div>
        <div
          className="section-claim__content-item__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin quis nunc vitae velit rutrum suscipit non et eros.
        </div>
      </div>
      <div
        className="section-claim__content__separator"
      >
        <img
          alt="after-blue"
          src="after-blue.png"
        />
      </div>
      <div
        className="section-claim__content-item"
      >
        <div
          className="section-claim__content-item__avatar __pink-circle"
        >
          2
        </div>
        <div
          className="section-claim__content-item__title"
        >
          Report Claim
        </div>
        <div
          className="section-claim__content-item__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin quis nunc vitae velit rutrum suscipit non et eros.
        </div>
      </div>
      <div
        className="section-claim__content__separator"
      >
        <img
          alt="after-pink"
          src="after-pink.png"
        />
      </div>
      <div
        className="section-claim__content-item"
      >
        <div
          className="section-claim__content-item__avatar __green-circle"
        >
          3
        </div>
        <div
          className="section-claim__content-item__title"
        >
          Final Settlement
        </div>
        <div
          className="section-claim__content-item__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin quis nunc vitae velit rutrum suscipit non et eros.
        </div>
      </div>
    </div>
  </div>
</div>
`;
